### DISCUSSION TITLE
Only display selected Annotations inside heatmap of "ClusterMapPlotter"

### DISCUSSION THREAD

**Discussion by GordianArnav** [DingWB/PyComplexHeatmap#54] [Q&A] [answered]
In ClusterMapPlotter, how to go about adding text other than float values, based on specific conditions relating to the actual heatmap values?

**Comment by DingWB** [comment]
Do you mean add text to the columns or rows annotations?

**Reply by GordianArnav** [reply]
For example, in ClusterMapPlotter, when "annot = True", the actual value is displayed inside the cells on the Main Heatmap.

How to go about displaying alternate text other than the actual value, based on some condition regarding the actual value?

Example, a use case would to mark cells that are significant in a heatmap

**Comment by DingWB** [comment, marked as answer]
It has already been implemented in PyComplexHeatmap. But I have never tested it.
You can pass a parameter `annot` to  `ClusterMapPlotter`, `annot` is also a dataframe.
Please try it and give me feedback.
Thanks

**Reply by GordianArnav** [reply]
Hi I was able to get it to work by passing a dataframe. Thanks!

**Reply by DingWB** [reply]
Could you please also paste your code and plot here? So that I can potentially put them on the documentation website to let more people know about this feature.

**Comment by DingWB** [comment]
Hello @GordianArnav  I updated the custom `annot` to support custom string annotation, please check https://dingwb.github.io/PyComplexHeatmap/build/html/notebooks/advanced_usage.html#Custom-annotation
